### PR TITLE
Initial support for time formats

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
This covers scalar time request and response.  Support for time within
structs will be added in a separate change.
Emit internal time helper types when a request and/or response contains
a time type.
Switched header format checks to use DateTimeSchema.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -64,5 +64,11 @@
...
       "directory": "./src",
     },
-  ]
+  ],
+  "go.useLanguageServer": true,
+  "go.languageServerExperimentalFeatures": {
+    "format": true,
+    "diagnostics": true,
+    "documentLink": true
+  }
```
Comment (catalinaperalta) [inline .vscode/settings.json]: Just wondering....shouldn't this file be in our gitignore?

Comment (jhendrixMSFT) [reply .vscode/settings.json]: Don't we use this to enable common eslint features?

Comment (catalinaperalta) [reply .vscode/settings.json]: Hmmm...well yes we could. I havent pushed my recent changes to this file though. But I could do it as well. So then we'll keep this in the repo until we merge track2 into master? And then delete it before then? Or would be keep it?

Comment (jhendrixMSFT) [reply .vscode/settings.json]: Well, if this is how we standardize on what eslint features to require then I think we'll have to keep it.

Comment (catalinaperalta) [reply .vscode/settings.json]: Yes that's true. Ok so we're keeping it 👍

**Pull request merged**